### PR TITLE
PCC53 Query the Spotify API

### PR DESCRIPTION
**Pull request by bryankimani**
ATTENTION: before clicking "Create Pull Request" please submit some meta data, thanks!

Difficulty level (1-10): []
Estimated time spent (hours): []
Completed (yes/no): []
I stretched my coding skills (if yes what did you learn?): []
Other feedback (what can we improve?): []

### CODE REVIEW

File: 53/bryankimani/artist_albums_names.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+        client_id=os.getenv("CLIENT_ID"), client_secret=os.getenv("CLIENT_SECRET")
+    )
+    sp = spotipy.Spotify(client_credentials_manager=client_credentials_manager)
+    return sp
+
+
+def get_artist_id():
+    # return the artist id from the entered artist name
+    artist_name = input("Please Enter Artist Name: ")
+    results_artist_name = spotify_auth().search(
```
Comment (pybites) [inline 53/bryankimani/artist_albums_names.py]: I would pull the auth out doing it once, then use the sp variable rather

File: 53/bryankimani/artist_albums_names.py [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+        results = spotify_auth().artist_albums(get_artist_id(), album_type="album")
+        for item in results["items"]:
+            print("Album Name          : " + item["name"])
+            print("Album URI           : " + item["uri"])
+            print("Album Release Date  : " + item["release_date"])
+            print("Album Total Track   : " + str(item["total_tracks"]))
+            print("Album Cover photo   : " + item["images"][0]["url"])
+            print()
+
+    except spotipy.client.SpotifyException:
```
Comment (pybites) [inline 53/bryankimani/artist_albums_names.py]: nice exception handling

File: 53/bryankimani/my_playlist.py [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+            print("   %d %32.32s %s" % (i, track["artists"][0]["name"], track["name"]))
+        except TypeError:
+            break
+
+
+def my_playlist():
+    username = input("Please enter your spotify username: ")
+    try:
+        playlists = spotify_auth().user_playlists(username)
+        for playlist in playlists["items"]:
```
Comment (pybites) [inline 53/bryankimani/my_playlist.py]: _flat is better than nested_, try making this try/exception smaller, by putting logic into helper functions

File: 53/bryankimani/top_tracks.py [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+import requests
+from dotenv import load_dotenv
+
+load_dotenv()
+
+username = input("Please enter your spotify username: ")
+artist_name = input("Please Enter Artist Name: ")
+
+
+def print_header():
```
Comment (pybites) [inline 53/bryankimani/top_tracks.py]: I see this code in various files, try to abstract it in a base class or single module

File: 53/bryankimani/top_tracks.py [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+    print("------------------------------------------")
+    print(" ARTIST TOP TRACKS THAT ARE IN MY PLAYLIST ")
+    print("------------------------------------------")
+    print()
+
+
+def spotify_auth():
```
Comment (pybites) [inline 53/bryankimani/top_tracks.py]: same comment as for `print_header`

**Comment by pybites** [comment]
Great work Bryan, keep it up!

File: 53/bryankimani/artist_albums_names.py (hunk 1 above)
Comment (bryankimani) [reply 53/bryankimani/artist_albums_names.py]: Great. Thanks for the pointer.

File: 53/bryankimani/top_tracks.py (hunk 4 above)
Comment (bryankimani) [reply 53/bryankimani/top_tracks.py]: Okay. Thanks. I will have change this

**Pull request merged**